scipy/ndimage/filters.py

```diff
@@ -1157,6 +1173,10 @@ def _rank_filter(input, rank, size=None, footprint=None, output=None,
     else:
         output = _ni_support._get_output(output, input)
         mode = _ni_support._extend_mode_to_code(mode)
+        if hasattr(mode, '__iter__') and not isinstance(mode, str):
```

Comment:
Probably should be one line above, before `mode = _ni_support._extend_mode_to_code(mode)` which will also raise